### PR TITLE
[504] Migrate production to AKS

### PR DESCRIPTION
**Pull request by smneal**
### Context

Change required to migrate production from GOV.UK PaaS to the Teacher Services Cloud AKS platform.

This builds an image where the production AKS environment uses the proper production domains and includes the DNS changes to point `www` (for find and publish) and `api` (for publish) to the new AKS environment.

### Changes proposed in this pull request

- Removal of production input and jobs from various GitHub action workflows and actions.
- Update the production_aks environment configuration to use the `www` and `api` domains. Previously `www3` and `api3`.
- Remove the existing CNAME records that are used by the GOV.UK PaaS environment
- Add the CNAME record for `www`, `api` to point to the previously provisioned front door endpoint. This may fail first time unless the existing CNAME deletion happens quickly enough. Just re-run it.
- Update the database backup and restore workflow which backs up prod, sanitises it and restores it to QA and staging to use the AKS production database and also postgres 14.7 so the pg_dump workers properly.
- Delete the scheduled PaaS to AKS backup and restore workflow.

### Guidance to review

- Tested database backup and sanitise job of workflow [here](https://github.com/DFE-Digital/publish-teacher-training/actions/runs/5680596414/job/15395031815).

### Checklist

- [ ] Make sure all information from the Trello card is in here
- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally
- [ ] Inform data insights team due to database changes

### CODE REVIEW

File: .github/actions/deploy_v2/action.yml [hunk 1]
```diff
@@ -49,7 +49,7 @@ runs:
           echo "DEPLOY_REF=${{ github.ref }}" >> $GITHUB_ENV
           hostname=$(jq -r '.additional_hostnames[0]' ${tf_vars_file})
           if [ ${app_environment} == "production" ]; then
-            echo "deploy_url=https://api3.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
+            echo "deploy_url=https://api.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
           else
             echo "deploy_url=https://${app_environment}2.api.publish-teacher-training-courses.service.gov.uk" >> $GITHUB_OUTPUT
```
Comment (RMcVelia) [inline .github/actions/deploy_v2/action.yml:54]: ${app_environment}2.api to ${app_environment}.api ?

**Comment by RMcVelia** [comment]
FYI, for register I cleaned up config/settings and config/environments straight after all envs migrated, to get rid of duplicate files.
i.e. https://github.com/DFE-Digital/register-trainee-teachers/pull/3445
or keep _aks and delete the other ones.

File: terraform/custom_domains/environment_domains/workspace_variables/find_production.tfvars.json [hunk 2]
```diff
@@ -4,6 +4,7 @@
       "front_door_name": "s189p01-ftt-svc-domains-fd",
       "resource_group_name": "s189p01-fttdomains-rg",
       "domains": [
+        "www",
         "www3"
```
[the remark this answers is not in the document]
Comment (smneal) [reply terraform/custom_domains/environment_domains/workspace_variables/find_production.tfvars.json:8]: I have been removing these after the migration as tidy-up to reduce the time waiting for the resources to be destroyed during the actual migration window.

File: .github/actions/deploy_v2/action.yml (hunk 1 above)
Comment (smneal) [reply .github/actions/deploy_v2/action.yml:54]: Good shout

File: .github/workflows/build-and-deploy.yml [hunk 3]
```diff
@@ -437,7 +414,7 @@ jobs:
       name: sandbox_aks
       url: ${{ steps.deploy_sandbox_v2.outputs.deploy-url }}
     if: ${{ success() && github.ref == 'refs/heads/main' }}
-    needs: [deploy-aks-production,deploy-production]
+    needs: [deploy-aks-production]
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
```
[the remark this answers is not in the document]
Comment (smneal) [reply .github/workflows/build-and-deploy.yml:378]: Updated this based on the comments

**Comment by smneal** [comment]
> FYI, for register I cleaned up config/settings and config/environments straight after all envs migrated, to get rid of duplicate files. i.e. [DFE-Digital/register-trainee-teachers#3445](https://github.com/DFE-Digital/register-trainee-teachers/pull/3445) or keep _aks and delete the other ones.

Thanks Rick. I've added this to the post-migration activity on the migration plan.

**Pull request merged**